bindings/c/manifoldc.cpp

```diff
@@ -0,0 +1,611 @@
+#include <manifold.h>
+#include <manifoldc.h>
+#include <meshIO.h>
+#include <public.h>
+#include <sdf.h>
+
+#include <algorithm>
+#include <conv.hh>
+#include <cstddef>
+#include <cstring>
+#include <functional>
+#include <optional>
+#include <vector>
+
+#include "box.cpp"
+#include "glm/ext/matrix_float3x3.hpp"
+#include "glm/ext/matrix_float4x3.hpp"
+#include "glm/ext/vector_float3.hpp"
+#include "glm/ext/vector_int3.hpp"
+#include "include/conv.hh"
+#include "include/types.h"
+#include "meshio.cpp"
+#include "types.h"
+
+using namespace manifold;
+
+namespace {
+ManifoldMesh *level_set(void *mem, float (*sdf)(float, float, float),
+                        ManifoldBox *bounds, float edge_length, float level,
+                        bool seq) {
+  // typing with std::function rather than auto compiles when CUDA is on,
+  // passing it into GPU (and crashing) is avoided dynamically in `sdf.h`
+  std::function<float(glm::vec3)> fun = [sdf](glm::vec3 v) {
+    return (sdf(v.x, v.y, v.z));
+  };
+  auto pol = seq ? std::make_optional(Seq) : std::nullopt;
```

Comment:
Is there a way we can void this bool parameter by making `LevelSet` choose the `Seq` policy when the template parameter is not a Functor? Curious of @pca006132 knows any magic here.

Reply:
`ParSeq` is fine for `std::function` though generally, it is only a problem from particular host languages that might be using the C bindings, so I wanted to provide a way to lock in sequential usage of the closure without adding another separate named method to the API.

Reply:
Ah, okay I see now.